voyager-screenmodel/src/commonMain/kotlin/cafe/adriel/voyager/core/model/NavigatorScreenModel.kt

```diff
@@ -23,6 +29,53 @@ public inline fun <reified T : ScreenModel> Navigator.rememberNavigatorScreenMod
     }
 }
 
+/**
+ * Lookup for a [ScreenModel] that was remembered in the current [LocalNavigator] or in your parents.
+ * If an instance is not found, an [IllegalStateException] will be thrown
+ *
+ * @param tag A custom tag used to "tag" the [ScreenModel]
+ * @throws IllegalStateException if no [ScreenModel] is found in [LocalNavigator] or in your parents
+ *
+ * @return The [ScreenModel] instance found
+ */
+@Suppress("UnusedReceiverParameter")
+@Throws(IllegalStateException::class)
+@Composable
+public inline fun <reified T : ScreenModel> Screen.navigatorScreenModel(
+    tag: String? = null
+): T {
+    var navigator = LocalNavigator.currentOrThrow
+    var screenModel: T? by remember(tag) { mutableStateOf(null) }
+    while (screenModel == null) {
+        screenModel = ScreenModelStore.getOrNull(
+            holderKey = navigator.key,
+            tag = tag
+        )
+        navigator = navigator.parent ?: break
+    }
+    return checkNotNull(screenModel) {
+        "${T::class} was not found in $navigator and it parents"
+    }
+}
+
+/**
+ * Remember a [ScreenModel] that will be scoped to current [LocalNavigator]
+ *
+ * @param tag A custom tag used to "tag" the [ScreenModel]
+ * @param factory A function to create a new instance if one is not remembered yet
+ *
+ * @return The [ScreenModel] instance
+ */
+@Suppress("UnusedReceiverParameter")
+@Composable
+public inline fun <reified T : ScreenModel> Screen.rememberNavigatorScreenModel(
+    tag: String? = null,
+    noinline factory: @DisallowComposableCalls () -> T
+): T {
+    val navigator = LocalNavigator.currentOrThrow
+    return navigator.rememberNavigatorScreenModel(tag = tag, factory = factory)
+}
```

Comment:
This API is no longer required, voyager has built in APIs navigator level scoped screen model